File: src/main/java/com/uiuc/cs498/netpromoter/analyzer/busobj/TweetAnalysis.java
package com.uiuc.cs498.netpromoter.analyzer.busobj;

import java.util.HashMap;
import java.util.Map;

import org.apache.spark.mllib.linalg.Vector;

public class TweetAnalysis {

	private String tweetText;
	private Integer score;
	private Vector probabilities;
	private Map<String, Integer> counts = new HashMap<>();
	
	public TweetAnalysis(){}

	public String getTweetText() {
		return tweetText;
	}

	public TweetAnalysis setTweetText(String tweetText) {
		this.tweetText = tweetText;
		return this;
	}

	public Integer getScore() {
		return score;
	}

	public TweetAnalysis setScore(Integer score) {
		this.score = score;
		return this;
	}

	public Vector getProbabilities() {
		return probabilities;
	}

	public TweetAnalysis setProbabilities(Vector probabilities) {
		this.probabilities = probabilities;
		return this;
	}

	public Map<String, Integer> getCounts() {
		return counts;
	}

	public TweetAnalysis setCounts(Map<String, Integer> counts) {
		this.counts = counts;
		return this;
	}
	
	
}

File: src/main/java/com/uiuc/cs498/netpromoter/analyzer/busobj/TweetAnalysis.scala
package com.uiuc.cs498.netpromoter.analyzer.busobj;

import java.util.HashMap;
import java.util.Map;

import org.apache.spark.mllib.linalg.Vector;

class TweetAnalysis() {

	var tweetText: String = ""
	var score : Integer = -1
	var probabilities : Vector = null
	var counts : Map[String, Integer] = new HashMap();
	

	def getTweetText(): String = {
		tweetText
	}

	def tweetText(tweetText: String): TweetAnalysis = {
		this.tweetText = tweetText
		this
	}

	def score(score :Integer):TweetAnalysis = {
		this.score = score
		this
	}

	def getProbabilities(): Vector = {
		probabilities
	}

	def probabilities(probabilities: Vector): TweetAnalysis = {
		this.probabilities = probabilities
		this
	}

	def getCounts() : Map[String, Integer] ={
		counts
	}

	def setCounts(counts :Map[String, Integer]): TweetAnalysis = {
		this.counts = counts
		this
	}
}
